app/include/zmk/events/battery_state_changed.h

```diff
@@ -14,4 +14,11 @@ struct zmk_battery_state_changed {
     uint8_t state_of_charge;
 };
 
-ZMK_EVENT_DECLARE(zmk_battery_state_changed);
\ No newline at end of file
+ZMK_EVENT_DECLARE(zmk_battery_state_changed);
+
+struct zmk_peripheral_battery_state_changed {
```

Comment:
Hmm.... We already have the concept of position events having a "source" (see `app/include/zmk/events/position_state_changed.h`). If we're now going to have yet another event that is either from local or peripheral device, we might want to pull that out in a more generalized place, and simply supplement the existin battery state changed event w/ a `source` field as well. Thoughts?

Reply:
I agree, I need some time to understand to how that's implemented/used.